<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>make_rule</class>
 <widget class="QDialog" name="make_rule">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>650</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1100</width>
    <height>650</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1100</width>
    <height>650</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Создание нового правила</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QFrame" name="frame">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>10</y>
     <width>741</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"> QFrame {
     background:  rgb(225, 225, 225);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QTextEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QTextEdit" name="edt_name_rule">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>10</y>
      <width>591</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="verticalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_Name_Rule">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Имя правила</string>
    </property>
   </widget>
  </widget>
  <widget class="QFrame" name="frame_2">
   <property name="geometry">
    <rect>
     <x>780</x>
     <y>10</y>
     <width>251</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"> QFrame {
     background:  rgb(225, 225, 225);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QComboBox" name="cbx_protocol">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>10</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>TCP</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>UDP</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="lbl_protocol">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Протокол</string>
    </property>
   </widget>
  </widget>
  <widget class="QFrame" name="frame_3">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>220</y>
     <width>1001</width>
     <height>411</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"> QFrame {
     background:  rgb(225, 225, 225);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }

</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLabel" name="lbl_Filters">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>50</y>
      <width>351</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Список фильтров, образующих Правило</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tbl_Filters">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>80</y>
      <width>931</width>
      <height>241</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QTableView QTableCornerButton::section {
background-color: rgb(108, 108, 108);
}

QHeaderView:section{
	 background-color: rgb(108, 108, 108);
    
	 color: white;
     margin:2px;
	 border-radius : 25px;
 } 
QHeaderView{
	 background-color: rgb(108, 108, 108);
	 color: white;
	 border: 1px solid #6c6c6c;
	 border-radius : 25px;
 }
QTableWidget{
	border: 2px solid   rgb(108, 108, 108);
    background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
      
    background-color: rgb(255, 255, 255);
}
</string>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarAlwaysOff</enum>
    </property>
    <attribute name="verticalHeaderCascadingSectionResizes">
     <bool>false</bool>
    </attribute>
   </widget>
   <widget class="QPushButton" name="btnAddFilter">
    <property name="geometry">
     <rect>
      <x>958</x>
      <y>80</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="toolTip">
     <string>Добавить фильтр в правило</string>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QFrame" name="frame_8">
    <property name="geometry">
     <rect>
      <x>351</x>
      <y>331</y>
      <width>321</width>
      <height>61</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"> QFrame {
    
	 background: rgb(207, 207, 207);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>-40</x>
       <y>0</y>
       <width>405</width>
       <height>58</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QToolButton" name="btn_Rule_OK">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>Принять изменения</string>
        </property>
        <property name="text">
         <string>Принять </string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/icons/check.png</normaloff>:/icons/check.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextBesideIcon</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolButton" name="btn_Rule_Cancel">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="toolTip">
         <string>Отменить изменения</string>
        </property>
        <property name="text">
         <string>Отмена  </string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/icons/cancel.png</normaloff>:/icons/cancel.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>25</width>
          <height>25</height>
         </size>
        </property>
        <property name="toolButtonStyle">
         <enum>Qt::ToolButtonTextBesideIcon</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
  </widget>
  <widget class="QFrame" name="frame_4">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>70</y>
     <width>461</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"> QFrame {
     background:  rgb(225, 225, 225);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLabel" name="lbl_port_">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>15</y>
      <width>32</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_port">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Порт</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="edt_Port_end">
    <property name="geometry">
     <rect>
      <x>343</x>
      <y>10</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLineEdit" name="edt_Port_begin">
    <property name="geometry">
     <rect>
      <x>263</x>
      <y>10</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QComboBox" name="cbx_kind_port">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>10</y>
      <width>161</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>161</width>
      <height>0</height>
     </size>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>по умолчанию</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>источника(номер)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>источника(диапазон)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>назначения(номер)</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>назначения(диапазон)</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QFrame" name="frame_6">
   <property name="geometry">
    <rect>
     <x>570</x>
     <y>130</y>
     <width>461</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"> QFrame {
     background:  rgb(225, 225, 225);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QComboBox" name="cbx_Packs_Action">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>10</y>
      <width>151</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>пропускать</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>блокировать</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>перенаправлять</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="lbl_Packs">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>20</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Пакеты</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cbx_Packs_Type">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>10</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>по умолчанию</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>входящие</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>исходящие</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QFrame" name="frame_5">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>70</y>
     <width>531</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"> QFrame {
     background:  rgb(225, 225, 225);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLineEdit" name="edt_IP_end">
    <property name="geometry">
     <rect>
      <x>431</x>
      <y>10</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_IP">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>IP</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="edt_IP_begin">
    <property name="geometry">
     <rect>
      <x>298</x>
      <y>10</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QComboBox" name="cbx_IP">
    <property name="geometry">
     <rect>
      <x>168</x>
      <y>10</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>по умолчанию</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>адрес</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>диапазон</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="lbl_IP_">
    <property name="geometry">
     <rect>
      <x>398</x>
      <y>20</y>
      <width>32</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cbx_kind_IP">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>10</y>
      <width>91</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>источника</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>назначения</string>
     </property>
    </item>
   </widget>
  </widget>
  <widget class="QFrame" name="frame_7">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>130</y>
     <width>531</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true"> QFrame {
     background:  rgb(225, 225, 225);
     border: 2px solid #C4C4C3;
     border-bottom-color: #C2C7CB; /* такой же цвет как у панели */
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }
QLabel {
 border-color: rgb(225, 225, 225);
}
QLineEdit {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                 stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
     border: 2px solid #C4C4C3;     
     background-color: rgb(255, 255, 255);
     border-radius: 4px;
     min-width: 8ex;
     padding: 2px;
 }</string>
   </property>
   <property name="frameShape">
    <enum>QFrame::StyledPanel</enum>
   </property>
   <property name="frameShadow">
    <enum>QFrame::Raised</enum>
   </property>
   <widget class="QLabel" name="lbl_Process">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Процесс</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cbx_Process_kind">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>10</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <item>
     <property name="text">
      <string>по умолчанию</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Название</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Название [PID]</string>
     </property>
    </item>
   </widget>
   <widget class="QComboBox" name="cbx_Process">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>10</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLineEdit" name="edt_PID">
    <property name="geometry">
     <rect>
      <x>458</x>
      <y>10</y>
      <width>51</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
   </widget>
  </widget>
 </widget>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections/>
</ui>
